tests/unit/lib/cookiecutter/test_template.py

```diff
@@ -0,0 +1,155 @@
+from unittest import TestCase
+from unittest.mock import patch, Mock, MagicMock
+from samcli.commands.exceptions import UserException
+from samcli.lib.cookiecutter.template import Template
+from samcli.lib.cookiecutter.exceptions import (
+    GenerateProjectFailedError,
+    InvalidLocationError,
+    PreprocessingError,
+    PostprocessingError,
+)
+from cookiecutter.exceptions import RepositoryNotFound, UnknownRepoType
+
+
+class TestTemplate(TestCase):
+    _ANY_LOCATION = "any/path/to/cookiecutter/template"
+    _INTERACTIVE_FLOW_MOCK = Mock()
+    _PREPROCESSOR_MOCK = Mock()
+    _POSTPROCESSOR_MOCK = Mock()
+    _PLUGIN_MOCK = Mock()
+    _ANY_INTERACTIVE_FLOW_CONTEXT = Mock()
+    _ANY_PLUGIN_INTERACTIVE_FLOW_CONTEXT = Mock()
+    _ANY_PROCESSOR_CONTEXT = Mock()
+
+    @patch("samcli.lib.cookiecutter.interactive_flow")
+    @patch("samcli.lib.cookiecutter.processor")
+    @patch("samcli.lib.cookiecutter.processor")
+    @patch("samcli.lib.cookiecutter.plugin")
+    def test_creating_a_template(self, mock_plugin, mock_preprocessor, mock_postprocessor, mock_interactive_flow):
```

Comment:
Could we add one more test where the supplied plugin already has pre-processors of its own?

Reply:
I already made this test for the interactive-flow(Template with direct interactive flow and a plugin's interactive flow) preprocessing and postprocessing are just the same concept, but no worries let me add them

Reply:
Oh, I actually have them [here](https://github.com/aws/aws-sam-cli/pull/2709/files#diff-3e916e76ac8ee3165271ce7efc4cf30d72710051b5987971c987e92de2f1c551R119-R123)

Reply:
Yep, understand the concept being the same, just want to ensure that both processors on the plugin and template are run.